### PR TITLE
Added numpy average frontend and test

### PR DESCRIPTION
**Pull request by shireenchand**
Issue number - https://github.com/unifyai/ivy/issues/6739

### CODE REVIEW

**Comment by shireenchand** [comment]
Hey @AnnaTz, can you please review the PR and let me know if any changes are required
Thanks!

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py [hunk 1]
```diff
@@ -162,3 +162,48 @@ def test_numpy_std(
...
     )
+
+
+#average
+@handle_frontend_test(
+    fn_tree="numpy.average",
+    dtype_and_a=statistical_dtype_values(function="average"),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.numpy.average"
+    ),
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py:173]: The current test handlers should not be given the `num_positional_args`. Could you please remove it?

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py [hunk 2]
```diff
@@ -162,3 +162,48 @@ def test_numpy_std(
...
+def test_numpy_average(
+        dtype_and_a,
+        as_variable,
+        with_out,
+        native_array,
+        num_positional_args,
+        dtype_and_x,
+        frontend,
+        fn_tree,
+        keep_dims,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py:186]: You should also add `on_device` to the arguments and pass it to `test_frontend_function` below, as done for example [here](https://github.com/unifyai/ivy/blob/2ae5f1e6b6ffd10e95c8e8c68f6747dd24cd4f0a/ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_decompositions.py#L27).

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py [hunk 3]
```diff
@@ -162,3 +162,48 @@ def test_numpy_std(
...
+        with_out=with_out,
+        native_array_flags=native_array,
+        num_positional_args=num_positional_args,
+        a=a[0],
+        weights=xs[0],
+        axis=axis,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        keepdims=keep_dims,
+        test_values=False,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py:207]: Why have you chosen to set `test_values=False`?

File: ivy/functional/frontends/numpy/statistics/averages_and_variances.py [hunk 4]
```diff
@@ -84,3 +84,48 @@ def std(
...
+    axis = tuple(axis) if isinstance(axis, list) else axis
+    global avg
+    avg = 0
+
+    if keepdims is None:
+        keepdims_kw = {}
+    else:
+        keepdims_kw = {'keepdims': keepdims}
+
+    # a = ivy.array(a)
```
Comment (AnnaTz) [inline ivy/functional/frontends/numpy/statistics/averages_and_variances.py:109]: You can clean up this comment.

**Review by AnnaTz — CHANGES_REQUESTED** [review]
Also, please have a look at `lint / check-formatting (pull_request)` > `Details`. There are some formatting errors that I think are caused by your PR.

**Comment by shireenchand** [comment]
Hi @AnnaTz, I have made the changes requested. Can you please check and let me know
Thanks!

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py [hunk 5]
```diff
@@ -162,3 +162,45 @@ def test_numpy_std(
...
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        weights=xs[0],
+        axis=axis,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        keepdims=keep_dims,
+        test_values=True,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py:204]: You can remove `test_values` entirely as `True` is the default value.

**Review by AnnaTz — CHANGES_REQUESTED** [review]
Could you please merge master to your PR? The CI test checks are failing due to an error that has now been resolved.

**Comment by shireenchand** [comment]
Hi @AnnaTz,
I have merged master into the PR. Can you check and let me know
Thanks!

File: ivy/functional/frontends/numpy/statistics/averages_and_variances.py [hunk 6]
```diff
@@ -84,3 +84,47 @@ def std(
...
+            avg = ivy.sum(mul, axis=axis, **keepdims_kw) / weights_sum
+
+        if returned:
+            if weights_sum.shape != avg.shape:
+                weights_sum = ivy.broadcast_to(weights_sum, avg.shape).copy()
+            return avg, weights_sum
+        else:
+            return avg
+    except ZeroDivisionError:
+        return 0
```
Comment (AnnaTz) [inline ivy/functional/frontends/numpy/statistics/averages_and_variances.py:130]: You need to make sure the test function does not cause the ZeroDivisionError.

**Review by AnnaTz — CHANGES_REQUESTED** [review]
Currently `test_numpy_average` has some failures you need to fix. To see them, please click on `Details` next to the failing `test-frontend-numpy / run-nightly-tests (pull_request)`, then click on `Run Frontend Tests` to expand it, and search for 'test_numpy_average'.

**Comment by shireenchand** [comment]
Hi @AnnaTz, I have handled the ZeroDivisionError as well as the infinity error. Can you please check
Thanks!

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py [hunk 7]
```diff
@@ -160,3 +160,49 @@ def test_numpy_std(
...
     )
+
+
+# average
+@handle_frontend_test(
+    fn_tree="numpy.average",
+    dtype_and_a=statistical_dtype_values(function="average"),
+    dtype_and_x=statistical_dtype_values(function="average"),
+    keep_dims=st.booleans(),
+)
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py:171]: Just noticed the `returned` argument is not tested. Could you please add it here?

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py [hunk 8]
```diff
@@ -752,7 +752,7 @@ def array_values(
                         ),
                         max_value=float_of(max_value, floats_info[dtype]["width"]),
                         allow_nan=allow_nan,
-                        allow_subnormal=allow_subnormal,
+                        # allow_subnormal=allow_subnormal,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:755]: Is there a reason you commented out these lines here?

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Review by AnnaTz — APPROVED** [review]
Everything looks good to me now. Ready to be merged!

**Pull request merged**